include/git2/worktree.h

```diff
@@ -86,6 +86,11 @@ typedef struct git_worktree_add_options {
 
 	int lock; /**< lock newly created worktree */
 	git_reference *ref; /**< reference to use for the new worktree HEAD */
+
+	/**
+	 * Options for the checkout.
+	 */
+	git_checkout_options *coopts;
```

Comment:
These should be a struct within the struct, not a pointer to another options struct.  See for example `git_submodule_update_options`.

```suggestion
	git_checkout_options checkout_opts;
```

That means adding default checkout options to `GIT_WORKTREE_ADD_OPTIONS_INIT`.

Curious why `FORCE` is the default?  I know that you didn't introduce that code but I'm curious now that we're revisiting it.  What happens if this is `SAFE` instead?

Reply:
Without pointer I have trouble to check if the checkout options was given by the user or not.

But maybe we could remove the default `GIT_CHECKOUT_FORCE` and only use the checkout options as implemented in `git_submodule_updates_options`.

Reply:
You shouldn't need to check if a user gave us checkout options.  User should either provide a `NULL` worktree options - and we'll use defaults for those options (and those include defaults for checkout options).  Or they gave us worktree options that include checkout options, and we'll use them.

Users are required to provide the options that they expect.  The `GIT_*_OPTIONS_INIT` macros and the `git_*_options_init` functions are responsible for giving them good defaults.  They can also pass `NULL` to get these options.

We don't need to give users the option to pass a half-empty options structure by passing in a pointer to another options structure.  This makes FFI languages extremely difficult because they'd need to create an opts struct, pin it, and then put a pointer to that in the other opts structure.

We have a lot of patterns where opts structure include other opts structures; we should not deviate from the existing pattern.